spec/surfnperf_spec.js

```diff
@@ -1096,5 +1096,136 @@ define('spec/surfnperf_spec', [
+        } else {
+          expect(SurfNPerf.chromeLoadTimes()).toEqual(undefined);
+        }
+      });
+    });
+
+    describe('#msFirstPaint', function() {
+      it('returns a reference to window.performance.timing.msFirstPaint if the browser supports it, otherwise undefined', function() {
+        if(window.performance && window.performance.timing && window.performance.timing.msFirstPaint) {
+          expect(SurfNPerf.msFirstPaint()).toEqual(window.performance.timing.msFirstPaint);
+        } else {
+          expect(SurfNPerf.msFirstPaint()).toEqual(undefined);
+        }
+      });
+    });
+
+    describe('#getFirstPaint', function() {
+      describe('when the client is using Chrome', function() {
+        it('returns the time to first paint relative to navigation start time', function() {
+          spyOn(SurfNPerf, 'chromeLoadTimes').andReturn({
+            firstPaintTime: 1446480965.123456
+          });
+          spyOn(SurfNPerf, 'msFirstPaint').andReturn(undefined);
+          spyOn(SurfNPerf, 'performanceTiming').andReturn({
+            navigationStart: 123789.734
+          });
+
+          expect(SurfNPerf.getFirstPaint()).toEqual(SurfNPerf.duration('navigationStart', 'chromePaint'));
+        });
+      });
+
+      describe('when the client is using IE/Edge', function() {
+        it('returns the time to first paint relative to navigation start time', function() {
+          spyOn(SurfNPerf, 'msFirstPaint').andReturn(1776480965.123457);
+          spyOn(SurfNPerf, 'chromeLoadTimes').andReturn(undefined);
+          spyOn(SurfNPerf, 'performanceTiming').andReturn({
+            navigationStart: 123789.734
+          });
+
+          expect(SurfNPerf.getFirstPaint()).toEqual(SurfNPerf.duration('navigationStart', 'msPaint'));
+        });
+      });
+
+      describe('when the client is using browser without Time to First Paint built in support', function() {
+        it('returns null since the browser does not have built in support', function() {
+          spyOn(SurfNPerf, 'chromeLoadTimes').andReturn(undefined);
+          spyOn(SurfNPerf, 'msFirstPaint').andReturn(undefined);
+          spyOn(SurfNPerf, 'performanceTiming').andReturn({
+            navigationStart: 123789.734
+          });
+ 
+          expect(SurfNPerf.getFirstPaint()).toEqual(null);
+        });
+      });
+    });
+
+    describe("#getFirstPaintFrame", function() {
+      describe('when the client does not support window.requestAnimationFrame', function() {
+        it('returns null', function() {
+          expect(SurfNPerf.getFirstPaintFrame()).toEqual(null);
+        });
+      });
+
+      describe('when the client supports window.requestAnimationFrame but it has not yet been called', function() {
+        it('returns undefined', function() {
+          if(!window.requestAnimationFrame) {
+            window.requestAnimationFrame = function() {};
+          }
+          expect(SurfNPerf.getFirstPaintFrame()).toEqual(undefined);
```

Comment:
after looking at the code, I think my original `undefined` return requirement was too much, so just make this expect that
```js
expect(isNaN(SurfNPerf.getFirstPaintFrame())).toBe(true);
```